frontend/src/lib/v2/CompareUtils.ts

```diff
@@ -55,6 +56,95 @@ export interface RunArtifactData {
   artifactCount: number;
 }
 
+export interface NameId {
+  name: string;
+  id: string;
+}
+
+// The provided name takes the format of "<type> ID #<number>" if the display name is not available.
+export interface FullArtifactPath {
+  run: NameId;
+  execution: NameId;
+  artifact: NameId;
+}
+
+export interface RocCurveArtifactData {
+  validLinkedArtifacts: LinkedArtifact[];
+  fullArtifactPathMap: { [key: string]: FullArtifactPath };
+}
+
+export const getRocCurveId = (linkedArtifact: LinkedArtifact): string =>
+  `${linkedArtifact.event.getExecutionId()}-${linkedArtifact.event.getArtifactId()}`;
+
+// Form an array which holds all valid ROC Curve linked artifacts.
+export const getValidRocCurveLinkedArtifacts = (
+  rocCurveRunArtifacts: RunArtifact[],
+): RocCurveArtifactData => {
+  const fullArtifactPathMap: { [key: string]: FullArtifactPath } = {};
+  const validLinkedArtifacts = flatMapDeep(
+    rocCurveRunArtifacts.map(runArtifact =>
+      runArtifact.executionArtifacts.map(executionArtifact => {
+        return getValidArtifacts(runArtifact, executionArtifact, fullArtifactPathMap);
+      }),
+    ),
+  );
+  return {
+    validLinkedArtifacts,
+    fullArtifactPathMap,
+  };
+};
+
+// Get the valid ROC Curve linked artifacts (those which have confidence metrics data),
+// and update the full artifact name map (run, execution, and artifact names are saved).
+const getValidArtifacts = (
+  runArtifact: RunArtifact,
+  executionArtifact: ExecutionArtifact,
+  fullArtifactPathMap: { [key: string]: FullArtifactPath },
+): LinkedArtifact[] => {
+  const validLinkedArtifacts: LinkedArtifact[] = [];
+  executionArtifact.linkedArtifacts.forEach(linkedArtifact => {
+    const customProperties = linkedArtifact.artifact.getCustomPropertiesMap();
+    const confidenceMetrics = customProperties
+      .get('confidenceMetrics')
+      ?.getStructValue()
+      ?.toJavaScript();
+    if (confidenceMetrics) {
+      validLinkedArtifacts.push(linkedArtifact);
+
+      // Save the names and IDs for the run, execution, and linked artifact to a map.
+      fullArtifactPathMap[getRocCurveId(linkedArtifact)] = getFullArtifactPath(
+        runArtifact.run,
+        executionArtifact.execution,
+        linkedArtifact,
+      );
+    }
+  });
+
+  return validLinkedArtifacts;
+};
+
+// This path is used to populate the ROC Curve filter table data.
+const getFullArtifactPath = (
+  run: ApiRunDetail,
+  execution: Execution,
+  linkedArtifact: LinkedArtifact,
+): FullArtifactPath => ({
+  run: {
+    name: run.run?.name || `Run ID #${run.run!.id!}`,
+    id: run.run!.id!,
+  },
+  execution: {
+    name: getExecutionDisplayName(execution) || `Execution ID #${execution.getId().toString()}`,
+    id: execution.getId().toString(),
+  },
+  artifact: {
+    name:
+      getArtifactName(linkedArtifact) ||
+      `Artifact ID #${linkedArtifact.artifact.getId().toString()}`,
+    id: linkedArtifact.artifact.getId().toString(),
+  },
```

Comment:
The object name is substituted with the ID if the name is not present. Only the `run ID` is currently needed in this object so it can be linked to in the ROC Curve filter table, but the execution and artifact IDs are provided for uniformity (and in case those can also be linked to directly in the future).